scipy/optimize/_lsq/lsq_linear.py

```diff
@@ -142,48 +142,48 @@ def lsq_linear(A, b, bounds=(-np.inf, np.inf), method='trf', tol=1e-10,
 
     See Also
     --------
-    nnls : Linear least squares with non-negativity constraint
-    least_squares : Robust nonlinear least squares with bound constraints
-                    on independent variables.   
+    nnls : Linear least squares with non-negativity constraint.
+    least_squares : Nonlinear least squares with bounds on the variables.                    
 
     Notes
     -----
-    The algorithm first computes the unconstrained solution by
+    The algorithm first computes the unconstrained least-squares solution by
     `numpy.linalg.lstsq` or `scipy.sparse.linalg.lsmr` depending on
-    `lsq_solver`. This solution is returned as optimal if it fits within the
+    `lsq_solver`. This solution is returned as optimal if it lies within the
     bounds.
 
     Method 'trf' runs the adaptation of the algorithm described in [STIR]_ for
     a linear least-squares problem. The iterations are essentially the same as
-    in the nonlinear algorithm, but as the quadratic model is always accurate
-    we don't need to track or modify a trust-region radius. The line search
-    (backtracking) is used as a safety net when the step does not decrease
-    the cost function. Read more detailed description of the algorithm
-    in `scipy.optimize.least_squares`.
+    in the nonlinear least-squares algorithm, but as the quadratic function
+    model is always accurate, we don't need to track or modify the radius of
+    a trust region. The line search (backtracking) is used as a safety net
+    when a selected step does not decrease the cost function. Read more
+    detailed description of the algorithm in `scipy.optimize.least_squares`.
 
     Method 'bvls' runs a Python implementation of the algorithm described in
     [BVLS]_. The algorithm maintains active and free sets of variables, on
     each iteration chooses a new variable to move from the active set to the
-    free set and then solves an unconstrained least-squares problem on free
-    variables. This method gives very accurate solution but may require up to
-    n iterations. Additionally, an ad-hoc initialization procedure is
+    free set and then solves the unconstrained least-squares problem on free
+    variables. This algorithm is guaranteed to give an accurate solution
+    eventually, but may require the number of iterations up to the number
+    of variables. Additionally, an ad-hoc initialization procedure is
     implemented, that determines which variables to set free or active
     initially. It takes some number of iterations before actual BVLS starts,
-    but can significantly reduce the number of iterations required by BVLS.
+    but can significantly reduce the number of future iterations.
```

Comment:
I meant "iterations after initialization". Is "future" a bad word for that here?